packages/web/src/components/download-track-archive-modal/DownloadTrackArchiveModal.tsx

```diff
@@ -0,0 +1,138 @@
+import { useCallback, useEffect } from 'react'
+
+import {
+  useCancelStemsArchiveJob,
+  useDownloadTrackStems,
+  useGetStemsArchiveJobStatus
+} from '@audius/common/api'
+import { useDownloadTrackArchiveModal } from '@audius/common/store'
+import {
+  Modal,
+  ModalContent,
+  ModalHeader,
+  Text,
+  Flex,
+  LoadingSpinner,
+  IconFolder,
+  ModalTitle,
+  IconReceive,
+  Hint,
+  IconError,
+  TextLink
+} from '@audius/harmony'
+
+import { env } from 'services/env'
+
+const messages = {
+  title: 'Preparing Download',
+  zippingFiles: (count: number) => `Zipping files (${count})`,
+  error: 'Something went wrong. Please check your connection and try again.',
+  tryAgain: 'Try again.'
+}
+
+const triggerDownload = (url: string) => {
+  if (document) {
+    const link = document.createElement('a')
+    link.href = url
+    link.click()
+    link.remove()
+  } else {
+    throw new Error('No document found')
+  }
+}
+
+export const DownloadTrackArchiveModal = () => {
+  const {
+    data: { trackId, fileCount },
+    isOpen,
+    onClose,
+    onClosed
+  } = useDownloadTrackArchiveModal()
+
+  const {
+    mutate: downloadTrackStems,
+    isError: initiateDownloadFailed,
+    isPending: isStartingDownload,
+    data: { id: jobId } = {}
+  } = useDownloadTrackStems({
+    trackId
+  })
+
+  const { mutate: cancelStemsArchiveJob } = useCancelStemsArchiveJob()
+
+  const { data: jobState } = useGetStemsArchiveJobStatus({
+    jobId
+  })
+
+  const hasError =
+    !isStartingDownload &&
+    (initiateDownloadFailed || jobState?.state === 'failed')
+
+  useEffect(() => {
+    if (isOpen) {
+      downloadTrackStems()
+    }
```

Comment:
i think what you have may be nicer, but one option is to return () => {cancelStemsArchiveJob()}?

Reply:
It's a little subtle, but we don't want to cancel every time the modal closes, only if the user explicitly closes the modal. When it auto-dismisses after the zip is created and the download starts, sending a cancel request would potentially cause the server to try and delete the file while we are trying to download it.